### PR TITLE
Clear vmthread from thread object at shutdown

### PR DESCRIPTION
**Pull request by tajila**
Related to: eclipse-openj9/openj9#16659

### CODE REVIEW

**Comment by tajila** [comment]
jenkins test sanity alinux64 jdk17

File: jcl/src/java.base/share/classes/java/lang/Thread.java [hunk 1]
```diff
@@ -1550,29 +1550,31 @@ void uncaughtException(Throwable e) {
  * @see J9VMInternals#threadCleanup()
  */
 void exit() {
+	try {
 /*[IF JAVA_SPEC_VERSION >= 14]*/
 	/* Refresh deadInterrupt value so it is accurate when thread reference is removed. */
 	deadInterrupt = interrupted();
```
Comment (JasonFengJ9) [inline jcl/src/java.base/share/classes/java/lang/Thread.java:1556]: The two lines above need to match the indention.

File: jcl/src/java.base/share/classes/java/lang/Thread.java [hunk 2]
```diff
@@ -1550,29 +1550,31 @@ void uncaughtException(Throwable e) {
...
+		runnable = null;
+		inheritedAccessControlContext = null;
 
-	synchronized(lock) {
-		threadRef = Thread.NO_REF;				// So that isAlive() can work
+		threadLocals = null;
+		inheritableThreadLocals = null;
+	} finally {
+		synchronized(lock) {
+			threadRef = Thread.NO_REF;		// So that isAlive() can work
```
Comment (JasonFengJ9) [inline jcl/src/java.base/share/classes/java/lang/Thread.java:1576]: Suggest moving the comment to its own line, and adding a period at the end.

**Review by JasonFengJ9 — APPROVED** [review]
LGTM, just minor formatting comments.

**Comment by tajila** [comment]
The aarch64 build https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK17_aarch64_linux/1730/

**Comment by tajila** [comment]
@gacholio please take a look

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
Testing already passed before formatting changes, merging.

**Pull request merged**